### PR TITLE
docs: Clean up OpenTelemetry docs

### PR DESCRIPTION
**Pull request by colleenmcginnis**
## Motivation/summary

Updates documentation related to OpenTelemetry based on issues found during our docs bug bash last week.

## Checklist

- [x] @colleenmcginnis update documentation:
  - [x] Remove most of the content in [Other execution environments](https://www.elastic.co/guide/en/apm/guide/current/open-telemetry-other-env.html) page and rename to "AWS Lambda support" (https://github.com/elastic/apm-server/commit/1cca6cdec279106c413888ad764ee39fa79d82eb)
  - [x] Update links on the OpenTelemetry [Limitations](https://www.elastic.co/guide/en/apm/guide/current/open-telemetry-known-limitations.html) page (https://github.com/elastic/apm-server/commit/8957ca5f5517ee5df68fc9e14c932474b4f602bc)
  - [x] Clarify that the code in [OpenTelemetry native support](https://www.elastic.co/guide/en/apm/guide/current/open-telemetry-direct.html) is just a Java example (https://github.com/elastic/apm-server/commit/51287a1d43d581b2f01d75b857b9af1c3d220398)
  - [x] Remove the TSVB example in [Collect metrics](https://www.elastic.co/guide/en/apm/guide/current/open-telemetry-collect-metrics.html) (https://github.com/elastic/apm-server/commit/ddf75de9d4e38b4c6948054c9e2066eab810ea40)
  - [x] Remove the resource attributes link in [OpenTelemetry native support](https://www.elastic.co/guide/en/apm/guide/current/open-telemetry-direct.html) (https://github.com/elastic/apm-server/commit/0f201673e5cd90c41527496932207775e4c87707)
- [x] @AlexanderWert review

## Related issues

* Closes https://github.com/elastic/observability-docs/issues/3291
* Closes https://github.com/elastic/observability-docs/issues/2196

### CODE REVIEW

**Comment by colleenmcginnis** [comment]
👋  @AlexanderWert @elastic/apm-server I'm looking for a review on these minor updates!

**Review by simitt — APPROVED** [review]
Would be good to also wait for review from @AlexanderWert as he requested some of the changes, but LGTM

**Review by bmorelli25 — APPROVED** [review]
LGTM as well. @AlexanderWert are you okay with these changes?

File: docs/otel-limitations.asciidoc
Comment (AlexanderWert) [inline docs/otel-limitations.asciidoc:36]:
```suggestion
The https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/v0.57.2/exporter/elasticexporter[OpenTelemetry Collector exporter for Elastic]
```

File: docs/otel-limitations.asciidoc
Comment (AlexanderWert) [inline docs/otel-limitations.asciidoc:40]:
```suggestion
The https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/elasticsearchexporter#elasticsearch-exporter[Elasticsearch exporter for the OpenTelemetry Collector]
```

**Review by AlexanderWert — DISMISSED** [review]
Two minor corrections regarding the links and naming of the Elastic / Elasticsearch exporters.

Otherwise, LGTM.

**Review by marclop — APPROVED** [review]

**Pull request merged**
